src/jdk.incubator.jextract/share/classes/jdk/internal/clang/SaveError.java

```diff
@@ -0,0 +1,61 @@
+/*
+ *  Copyright (c) 2020, Oracle and/or its affiliates. All rights reserved.
+ *  DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ *  This code is free software; you can redistribute it and/or modify it
+ *  under the terms of the GNU General Public License version 2 only, as
+ *  published by the Free Software Foundation.  Oracle designates this
+ *  particular file as subject to the "Classpath" exception as provided
+ *  by Oracle in the LICENSE file that accompanied this code.
+ *
+ *  This code is distributed in the hope that it will be useful, but WITHOUT
+ *  ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ *  FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ *  version 2 for more details (a copy is included in the LICENSE file that
+ *  accompanied this code).
+ *
+ *  You should have received a copy of the GNU General Public License version
+ *  2 along with this work; if not, write to the Free Software Foundation,
+ *  Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ *   Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ *  or visit www.oracle.com if you need additional information or have any
+ *  questions.
+ *
+ */
+package jdk.internal.clang;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.function.Function;
+
+import static java.util.stream.Collectors.toMap;
+import static jdk.internal.clang.libclang.Index_h.CXSaveError_InvalidTU;
+import static jdk.internal.clang.libclang.Index_h.CXSaveError_None;
+import static jdk.internal.clang.libclang.Index_h.CXSaveError_TranslationErrors;
+import static jdk.internal.clang.libclang.Index_h.CXSaveError_Unknown;
+
+public enum SaveError {
+    None(CXSaveError_None),
+    Unknown(CXSaveError_Unknown),
+    TranslationErrors(CXSaveError_TranslationErrors),
+    InvalidTU(CXSaveError_InvalidTU);
+
+    private final int code;
+
+    SaveError(int code) {
+        this.code = code;
+    }
+
+    public int code() {
+        return code;
+    }
+
+    private static final Map<Integer, SaveError> lookup = Arrays.stream(values())
+            .collect(toMap(SaveError::code, Function.identity()));
+
+    public static SaveError valueOf(int code) {
+        return lookup.computeIfAbsent(code, k -> { throw new NoSuchElementException("No SaveError with code: " + k); });
```

Comment:
Do we need this computeIfAbsent with lambda? All we do is to throw exception always if key is not found.

Reply:
`Map` doesn't have a `getOrThrow`, but `computeIfAbsent` effectively works the same way. The alternative would be `get` plus null check.